lib/ABI/FunctionType.cpp

```diff
@@ -154,6 +155,11 @@ class ConversionHelper {
     for (auto &Argument : *ArgumentList)
       Result.Arguments.insert(Argument);
 
+    auto StackArgumentList = convertStackArguments(Function.StackArgumentsType,
+                                                   Result.Arguments.size());
```

Comment:
Why is the number of arguments relevant?

Reply:
I don't understand your answer. I think you think I was implying something with my question that I was not implying `:)`

Reply:
Uhm, I'm more talking about the output, we're going from `RawFunctionType` to `CABIFunctionType`.

Let me try to explain myself better.

1. We're going from a `struct` representing all stack arguments (`RawFunctionType::StackArgumentsType`) to multiple arguments that will end up on the stack (`CABIFunctionType`). Right?
2. One option would be to pass a single `struct` representing all of the stack arguments, while you seem to be taking the more aggressive (and prettier!) route to add multiple arguments that will end up on the stack with the *exact same* layout as `RawFunctionType::StackArgumentsType`. Right?

What I'm saying is: if we can do this safely (i.e., no mismatches in layout), let's do it!

Reply:
Sounds good! I just asked because maybe you knew about some irregular alignment patterns under some architectures I haven't tested under just yet. If there's nothing obvious, it's worth a shot - worst case we can always go back to "preserving" the struct in those corner cases.

Reply:
That's great to hear! Be bold and test your boldness! `:)`